Merging img2vec node

cerashdan opened the pull request.

cerashdan committed:
adding modified files from previous branch

cerashdan committed:
updated requirement.txt

acmiyaguchi commented on ros/img2vec/CMakeLists.txt:
```diff
@@ -0,0 +1,207 @@
+# find_package(Boost REQUIRED COMPONENTS system)
+
+
+## Uncomment this if the package has a setup.py. This macro ensures
```
nit: remove all the comments that are uneeded.

acmiyaguchi commented on ros/img2vec/scripts/img2vec.py:
```diff
@@ -0,0 +1,74 @@
+from torch.autograd import Variable
+from PIL import Image
+ 
+class image_converter:
```
CamelCase is more Pythonic.

```suggestion
class ImageConverter:
```

acmiyaguchi commented on ros/img2vec/scripts/img2vec.py:
```diff
@@ -0,0 +1,74 @@
+      print(e)
+
+    image_embeddings = get_vector(cv_image)
+    print("embedding generetaed")
```
I would use the logging module instead, because you can turn up or down the verbosity with a flag. Printing things to the console can make programs slow when you do them many times a second.

acmiyaguchi commented on ros/img2vec/scripts/img2vec.py:
```diff
@@ -0,0 +1,74 @@
+    except CvBridgeError as e:
+      print(e)
+# source: https://becominghuman.ai/extract-a-feature-vector-for-any-image-with-pytorch-9717561d1d4c
+    def get_vector(image):
```
Make this a class method, its generally bad practice to put a definition inside of a definition.

cerashdan committed:
adding custom message type

cerashdan committed:
generated image embeddings

cerashdan committed:
published custom message with embeddings and header messages

cerashdan committed:
adding ros launch file and docs images

cerashdan commented:
Hi @acmiyaguchi , 

I created the following, please review: 

1. Img2vec node will take an image and publish image embeddings based on Res18 - Avgpooling layer of size 512 
2. The node has a launch file that could be run with the following parameters: 
```bash 
roslaunch --wait img2vec img2vec.launch camera_topic:=image_raw dest_topic:=image_embeddings
```
3. The ros launch command is included in the docker-compose.yaml and it should be launched with all other nodes.
4. The node publishes a custom image of type ImageEmbeddings.msg defined under img2vec pacakge. The custom message contains the standard ROS header and a flot64 array which contains the image embeddings
5. Adjusted from print statements to ROS Logging statements 
6. Added all required dependencies and followed dependencies convention.
7. Fixed minor comments above
8. Updated the README.md with diagrams and adjusted lines.

cerashdan committed:
adjusting node name

acmiyaguchi reviewed (approved):
There are some minor formatting issues. For future PRs, you should install the `pre-commit` hooks.
